.github/workflows/ci.yml

```diff
@@ -25,6 +25,10 @@ jobs:
       - name: Run unit tests
         run: >-
           npm run test:unit
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.0.1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
The default value of optional `fail_ci_if_error` input is false. If there is an upload error, do we consider this as CI error fail or do we ignore it?

Reply:
I think I'd be happier with this if I trusted Codecov to make sane assessments of the actual coverage.  I've had plenty of times when the missed coverage it reports is a mystery to me.  Plus you can refactor code and trigger identification of missed coverage when you're not in a position to be writing the tests right now.

On the other hand though, if failure does not stop you from merging with an override, there's no harm done.  I'll just be shaking my fist at it a little more vigourously when it tells me to do better.

Reply:
Ah yes, of course.  You're not failing because coverage is below expected levels.  You're failing because the upload didn't happen.  Which is a legitimate reason to want to re-run the tests.

Reply:
I would prefer we do not make coverage failure a code failure. I am not a fan of codecov. I have been stung many times where coverage has reduced by 0.1% and the test suite fails. In the end in some repos I removed it all together. I think our test coverage and code review are pretty good. If other developers like codecov then fair enough so I am happy for a compromise.

Reply:
Unless I got the wrong end of the stick @stevieing I believe the suggestion here is that the tests should fail if the upload of the results to Codecov fails.  Not that the tests should fail because Codecov didn't think we did enough coverage.

Reply:
BTW. Sometimes, there is no way we can, or we will, apply the suggestions of codecov. For example, it always complained about the module name V2 in Limber (Sequencescape API). I have found a way to override it in the code comments by using inline instructions to `reek` gem used by Codecov. You need to specify

`# :reek:CodeSmellNameToIgnore {Any additional params}`

```
The following is to make it stop complaining about V2
:reek:UncommunicativeModuleName
module Sequencescape::Api::V2
```

The following is to make stop complaining about using non-instance method as a private util.
```
 # :reek:UtilityFunction { public_methods_only: true }
    def study_required_number_of_cells_key(presenter)

```

Reply:
You are right. Thanks for pointing that out.